crates/rust-analyzer/src/bin/logger.rs

```diff
@@ -63,11 +64,20 @@ impl Log for Logger {
                 record.args(),
             ),
         }
+
+        if self.no_buffering {
+            self.flush();
+        }
     }
 
     fn flush(&self) {
-        if let Some(w) = &self.file {
-            let _ = w.lock().flush();
+        match &self.file {
+            Some(w) => {
+                let _ = w.lock().flush();
+            }
+            None => {
+                let _ = io::stderr().flush();
+            }
```

Comment:
Ahhhhhh....

This explains why logs misbehave for the last n months.

Let's just flush after every message unconditionally.

Reply:
> Let's just flush after every message unconditionally.

Sorry, but looks like a bad idea for me. It's super slow actually, for stderr too.
I think most of the time buffering not bad at all. And it's easy to add the flag if you need to.

Reply:
Hm, that's surprisng, I don't think it should be slow even with sync enabled, as most messages are rather long. Could you double-check that it's the flushing that is slow? 

But maybe I am missing smth windows-specific here...